### PR TITLE
Added isGlob function

### PR DESCRIPTION
**Pull request by dshubhadeep**
In reference to issue  #424

### CODE REVIEW

File: fs/glob.ts [hunk 1]
```diff
@@ -43,3 +43,36 @@ export interface GlobOptions {
 export function glob(glob: string, options: GlobOptions = {}): RegExp {
   return globrex(glob, options).regex;
 }
+
+/** Test whether the given string is a glob */
+export function isGlob(str: string): boolean {
+  const chars: any = { "{": "}", "(": ")", "[": "]" };
```
Comment (bartlomieju) [inline fs/glob.ts]: Can you use more specific type?

File: fs/glob_test.ts [hunk 2]
```diff
@@ -138,4 +138,120 @@ testWalk(
   }
 );
 
+test({
+  name: "isGlob: pattern to test",
+  fn(): void {
+    // should be true if valid glob pattern
+    assertEquals(isGlob("!foo.js"), true);
```
Comment (bartlomieju) [inline fs/glob_test.ts]: You can substitute it with `assert(isGlob("!foo.js"));`

File: fs/glob_test.ts [hunk 3]
```diff
@@ -138,4 +138,120 @@ testWalk(
   }
 );
 
+test({
+  name: "isGlob: pattern to test",
```
Comment (bartlomieju) [inline fs/glob_test.ts]: There should be test case testing that something is not glob `assert(!isGlob("foo.js"));`

File: fs/glob.ts (hunk 1 above)
Comment (zekth) [reply fs/glob.ts]: `Record<string,string>`

File: fs/glob_test.ts (hunk 3 above)
Comment (dshubhadeep) [reply fs/glob_test.ts]: The test cases include both valid and invalid glob patterns,
I have used these [test cases](https://github.com/micromatch/is-glob/blob/master/test.js).

**Comment by kitsonk** [comment]
Is this an fs thing or a string thing?  Glob patterns can be expressed in lots of other ways.  Yes, they are mainly used to match lists of files, but they can also be used with URI matching too.

File: fs/glob.ts [hunk 4]
```diff
@@ -43,3 +43,35 @@ export interface GlobOptions {
 export function glob(glob: string, options: GlobOptions = {}): RegExp {
   return globrex(glob, options).regex;
 }
+
+/** Test whether the given string is a glob */
+export function isGlob(str: string): boolean {
+  const chars: Record<string, string> = { "{": "}", "(": ")", "[": "]" };
+  const regex: RegExp = /\\(.)|(^!|\*|[\].+)]\?|\[[^\\\]]+\]|\{[^\\}]+\}|\(\?[:!=][^\\)]+\)|\([^|]+\|[^\\)]+\))/;
```
Comment (kt3k) [inline fs/glob.ts]: The linter seems complaining on this line.
```
/home/vsts/work/1/s/fs/glob.ts
  50:9  error  Type RegExp trivially inferred from a RegExp literal, remove type annotation  @typescript-eslint/no-inferrable-types
```

Comment (dshubhadeep) [reply fs/glob.ts]: Is it okay now ?

**Pull request merged**